test/sampler/test_povm_sampler_job.py

```diff
@@ -43,51 +47,116 @@ def test_initialization(self):
         cs_shots = 32
-        cs_job = povm_sampler.run([qc_random], shots=cs_shots, povm=cs_implementation)
-        result = cs_job.result()[0]
-        self.assertIsInstance(result, POVMPubResult)
+        with self.subTest("Result for a single PUB."):
+            cs_job = povm_sampler.run([qc_random], shots=cs_shots, povm=cs_implementation)
+            result = cs_job.result()
+            self.assertIsInstance(result, PrimitiveResult)
+            self.assertIsInstance(result[0], POVMPubResult)
+        with self.subTest("Result for multiple PUBs."):
+            cs_job = povm_sampler.run(
+                [qc_random, qc_random], shots=cs_shots, povm=cs_implementation
+            )
+            result = cs_job.result()
+            self.assertIsInstance(result, PrimitiveResult)
+            self.assertEqual(len(result), 2)
+            self.assertIsInstance(result[0], POVMPubResult)
+            self.assertIsInstance(result[1], POVMPubResult)
+        with self.subTest(
+            "Error raised if incompatible lengths of raw results and metadata."
+        ) and self.assertRaises(ValueError):
+            cs_job = povm_sampler.run(
+                [qc_random, qc_random], shots=cs_shots, povm=cs_implementation
+            )
+            cs_job.metadata.pop()
+            _ = cs_job.result()
 
     def test_recover_job(self):
         qc = QuantumCircuit(2)
         qc.h(0)
         qc.cx(0, 1)
 
         backend = FakeSherbrooke()
-        backend.set_options(seed_simulator=25)
-        pm = generate_preset_pass_manager(optimization_level=2, backend=backend)
+        backend.set_options(seed_simulator=self.RNG_SEED)
+        pm = generate_preset_pass_manager(
+            optimization_level=2, backend=backend, seed_transpiler=self.RNG_SEED
+        )
 
         qc_isa = pm.run(qc)
 
-        measurement = ClassicalShadows(2, seed_rng=13)
-        runtime_sampler = RuntimeSampler(backend=backend)
+        measurement = ClassicalShadows(2, seed_rng=self.RNG_SEED)
+        runtime_sampler = RuntimeSampler(mode=backend)
         povm_sampler = POVMSampler(runtime_sampler)
         job = povm_sampler.run(pubs=[qc_isa], shots=128, povm=measurement)
-        job.save_metadata(filename="saved_metadata.pkl")
         tmp = job.base_job
 
-        job_recovered = POVMSamplerJob.recover_job(filename="saved_metadata.pkl", base_job=tmp)
-        self.assertIsInstance(job_recovered, POVMSamplerJob)
-        result = job_recovered.result()
-        pub_result = result[0]
-        observable = SparsePauliOp(["II", "XX", "YY", "ZZ"], coeffs=[1, 1, -1, 1])
-        post_processor = POVMPostProcessor(pub_result)
-        exp_value, std = post_processor.get_expectation_value(observable)
-        self.assertAlmostEqual(exp_value, 3.53125)
-        self.assertAlmostEqual(std, 0.3590672895231641)
+        with self.subTest("Save job with specific filename."):
+            job.save_metadata(filename="saved_metadata.pkl")
```

Comment:
This generates a file on disk which will be left after the tests have completed.
Can you do smth along the following lines?
```python
with self.subTest("..."):
    filename = "..."
    job.save_metadata(filename=filename)
    # more code
    try:
        # run assertions: self.assertAlmostEqual
    except:  # catch anything
         pass
    finally:
        # delete filename (e.g. pathlib.Path(filename).unlink(missing_ok=True)
````
This will ensure that the file always gets cleaned up, no matter if the test passed or failed.
We should do the same whenever we may generate files on disk.